ParkingLotApiTest/ControllerTest/ParkingLotsControllerTests.cs

```diff
@@ -0,0 +1,249 @@
+            Assert.Equal(parkingLot, returnedParkingLot);
+
+            var getResponse = await client.GetAsync(createResponse.Headers.Location);
+            getResponse.EnsureSuccessStatusCode();
+            var newParkingLot = await DeserializeResponseBodyAsync<ParkingLotDto>(getResponse);
+            Assert.Equal(parkingLot, newParkingLot);
+        }
+
+        [Fact]
+        public async Task Should_not_create_parking_lot_when_add_new_parking_lot_with_negative_capacity()
+        {
+            var client = GetClient();
+            var parkingLot = SeedParkingLot();
+            parkingLot.Capacity = -1;
+            var parkingLotContent = SerializeRequestBody(parkingLot);
+
+            var createResponse = await client.PostAsync(RootUri, parkingLotContent);
+
+            Assert.False(createResponse.IsSuccessStatusCode);
+
+            var getResponse = await client.GetAsync(RootUri);
+            getResponse.EnsureSuccessStatusCode();
+            var allParkingLots = await DeserializeResponseBodyAsync<List<ParkingLotDto>>(getResponse);
+            Assert.Empty(allParkingLots);
+        }
+
+        [Fact]
+        public async Task Should_return_parking_lot_when_get_by_id()
+        {
+            var client = GetClient();
+            var parkingLot = SeedParkingLot();
+            var parkingLotContent = SerializeRequestBody(parkingLot);
+
+            var createResponse = await client.PostAsync(RootUri, parkingLotContent);
+            createResponse.EnsureSuccessStatusCode();
+            var getResponse = await client.GetAsync(createResponse.Headers.Location);
+
+            getResponse.EnsureSuccessStatusCode();
+            var newParkingLot = await DeserializeResponseBodyAsync<ParkingLotDto>(getResponse);
+            Assert.Equal(parkingLot, newParkingLot);
+        }
+
+        [Fact]
+        public async Task Should_return_all_parking_lot_when_get_all()
+        {
+            var client = GetClient();
+            var parkingLots = SeedParkingLots();
+            foreach (var lot in parkingLots)
+            {
+                var content = SerializeRequestBody(lot);
+                await client.PostAsync(RootUri, content);
+            }
+
+            var getResponse = await client.GetAsync(RootUri);
+
+            getResponse.EnsureSuccessStatusCode();
+            var returnedParkingLots = await DeserializeResponseBodyAsync<List<ParkingLotDto>>(getResponse);
+            Assert.Equal(2, returnedParkingLots.Count);
+            Assert.Equal(parkingLots[0], returnedParkingLots.First(lot => lot.Name == "first"));
+            Assert.Equal(parkingLots[1], returnedParkingLots.First(lot => lot.Name == "second"));
+        }
+
+        [Fact]
+        public async Task Should_return_parking_lots_in_pages_when_get_using_paging_query()
+        {
+            var client = GetClient();
+            var parkingLots = SeedParkingLots();
```

Comment:
maybe we should test 15 parkinglots because the requirement want 15 items in one page

Reply:
> maybe we should test 15 parkinglots because the requirement want 15 items in one page

agree, I will fix that later.

Reply:
> > maybe we should test 15 parkinglots because the requirement want 15 items in one page
> 
> agree, I will fix that later.

resolved.